docs/components/kafka.md

```diff
@@ -0,0 +1,78 @@
+# Kafka
+
+[https://kafka.apache.org/:](https://kafka.apache.org/)
+>Kafka® is used for building real-time data pipelines and streaming apps. It is horizontally scalable,
+fault-tolerant and wicked fast.
+
+Kafka is used by CommCare to provide stream processing capabilities for data changes. Changes to certain
+data models, such as cases and forms, are written to Kafka. Separate processes then read the changes
+from Kafka and process (pillows) them into Elasticsearch or other secondary databases.
+
+## Resources
+
+* [Command line tools](https://cwiki.apache.org/confluence/display/KAFKA/Replication+tools)
+
+## Dependancies
+* Java
+* Zookeeper
+
+## Setup
+
+Kafka is installed on each host in the *kafka* inventory group and Zookeeper is installed on each
+host in the *Zookeeper* inventory group.
+
+## Expanding the cluster
+
+1. Add new host to the inventory and install Kafka
+```
+$ commcare-cloud <env> deploy-stack --limit=<new kafka host>
+```
+
+2. Update localsettings
+```
+$ commcare-cloud <env> update-config
+```
+
+3. Move partitions to new node
+
+Follow the steps outlined [below](#move-partitions).
+
+## Useful commands
+All of the command below assume they are being run from the `/opt/kafka/bin/` path.
+
+### Show topic configuration
+```
+$ ./kafka-topics.sh --describe --zookeeper=<zookeeper host>:2181 --topic <topic>
+```
+
+### Add new partitions to topic
```

Comment:
Are we planning to have replication of topics and partitions?

Reply:
No plans for replication but I have tested out having multiple brokers